src/common/debug.ts

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+const usePrevious = (value: any, initialValue: any) => {
+  const ref = useRef(initialValue);
+  useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
+
+const useEffectDebugger = (effectHook: any, dependencies: any, dependencyNames: any = []) => {
```

Comment:
where are these being used?

Reply:
I used them to debug issues around creating multiple WS subscriptions, happy to remove them if you think they're not useful or confusing.

Reply:
we can keep them. Just curious